api/catalog/api/utils/validate_images.py

```diff
@@ -28,6 +30,28 @@ def _get_expiry(status, default):
     return config(f"LINK_VALIDATION_CACHE_EXPIRY__{status}", default=default, cast=int)
 
 
+async def _head(url: str, session: aiohttp.ClientSession) -> tuple[str, int]:
+    try:
+        async with session.head(url, timeout=2, allow_redirects=False) as response:
+            return url, response.status
+    except aiohttp.ClientError as exception:
+        _validation_failure(exception)
+        return url, -1
+
+
+# https://stackoverflow.com/q/55259755
+@async_to_sync
+async def _make_head_requests(urls: list[str]) -> list[tuple[str, int]]:
+    tasks = []
+    async with aiohttp.ClientSession(headers=HEADERS) as session:
+        for url in urls:
```

Comment:
Can this also be converted to a list comprehension, or do you think it makes it more difficult to understand?
`tasks = [asyncio.ensure_future(_head(url, session)) for url in urls`]